macros/helpers/prime_line.cfg

```diff
@@ -1,18 +1,46 @@
 [gcode_macro PRIMELINE]
 gcode:
-    # Macro parameters
+    # Base macro parameters
     {% set prime_line_length = params.LINE_LENGTH|default(printer["gcode_macro _USER_VARIABLES"].prime_line_length)|float %}
     {% set prime_line_purge_distance = params.PURGE_LENGTH|default(printer["gcode_macro _USER_VARIABLES"].prime_line_purge_distance)|float %}
     {% set prime_line_flowrate = params.FLOWRATE|default(printer["gcode_macro _USER_VARIABLES"].prime_line_flowrate)|float %}
-    {% if printer["gcode_macro _USER_VARIABLES"].prime_line_height %}
-      {% set prime_line_height = params.LINE_HEIGHT|default(printer["gcode_macro _USER_VARIABLES"].prime_line_height)|float %}
-    {% else %}
-      {% set prime_line_height = params.LINE_HEIGHT|default(0.6)|float %}
+    {% set prime_line_height = params.LINE_HEIGHT|default(printer["gcode_macro _USER_VARIABLES"].prime_line_height)|default(0.6)|float %}
+    {% set prime_line_adaptive = params.ADAPTIVE_MODE|default(True) %}
+    
+    # If the SIZE parameter is defined and not a dummy placeholder, we use it to do the adaptive bed mesh logic
+    {% set coordinatesFound = false %}
+    {% if params.SIZE is defined and params.SIZE != "0_0_0_0" %}
+        {% set xMinSpec, yMinSpec, xMaxSpec, yMaxSpec = params.SIZE.split('_')|map('trim')|map('int') %}
+        {% set coordinatesFound = true %}
+    {% elif printer.exclude_object is defined %}
```

Comment:
IMHO, using exclude_object to define the prime line introduces risks of collision. Although there's a lot of work being done in this area, not all slicers use exclude_object for brim, skirts and purge towers. 

But if you feel that this code should be maintained :
This code is already used in Adaptive_bed_mesh, would it be interesting to take it out of these 2 macros to introduce a mechanism to populate FL_SIZE whith exclude object during START_PRINT, for example or as reverse define an exclude_object with FL_SIZE ? Having in mind the compatibility with Klippe adaptive_mesh.

Reply:
This will be reworked later as I do plan to switch to the mainline adaptive bed mesh at some point when slicer will manage correctly the purge tower as an object. And I'll indeed work on defining fake objects definitions to be able to use it the same way in manual calibrations macros and make it transparent for an user.